code/game/objects/structures/kitchen_spike.dm

```diff
@@ -112,16 +114,76 @@
 				user << "<span class='danger'>The spike already has something on it; finish collecting its meat first!</span>"
 			return
 		if(istype(G.affecting, /mob/living/carbon/human))
-			user.changeNext_move(CLICK_CD_MELEE)
-			playsound(src.loc, "sound/effects/splat.ogg", 25, 1)
-			var/mob/living/carbon/human/H = G.affecting
-			H.visible_message("<span class='danger'>[user] slams [G.affecting] into the meat spike!</span>", "<span class='userdanger'>[user] slams you into the meat spike!</span>", "<span class='italics'>You hear a squishy wet noise.</span>")
-			H.adjustBruteLoss(20)
-			return
+			if(!occupied)
+				if(do_mob(user, src, 50))
+					if(occupied) //to prevent spam/queing up attacks
+						return
+					if(G.affecting.buckled)
+						return
+					var/mob/living/carbon/human/H = G.affecting
+					playsound(src.loc, "sound/effects/splat.ogg", 25, 1)
+					H.visible_message("<span class='danger'>[user] slams [G.affecting] onto the meat spike!</span>", "<span class='userdanger'>[user] slams you onto the meat spike!</span>", "<span class='italics'>You hear a squishy wet noise.</span>")
+					H.loc = src.loc
+					H.emote("scream")
+					var/turf/pos = get_turf(H)
+					pos.add_blood_floor(H)
+					H.adjustBruteLoss(30)
+					var/matrix/m120 = matrix()
+					m120.Turn(180)
+					animate(H, transform = m120, time = 3)
+					H.pixel_y = H.get_standard_pixel_y_offset(180)
+					occupied = TRUE
+					H.buckled = src
+					H.dir = 2
+					buckled_mob = H
+					return
 		user << "<span class='danger'>You can't use that on the spike!</span>"
 		return
 	..()
 
+/obj/structure/kitchenspike/user_buckle_mob(mob/living/M, mob/living/user) //Don't want them getting put on the rack other than by spiking
+	return
+
+/obj/structure/kitchenspike/user_unbuckle_mob(mob/living/carbon/human/user)
+	if(buckled_mob && buckled_mob.buckled == src)
+		var/mob/living/M = buckled_mob
+		if(M != user)
+			M.visible_message(\
+				"[user.name] tries to pull [M.name] free of the [src]!",\
+				"<span class='notice'>[user.name] is trying to pull you off the [src], opening up fresh wounds!</span>",\
+				"<span class='italics'>You hear a squishy wet noise.</span>")
+			M.adjustBruteLoss(20)

```

Comment:
Probably should add a cooldown or else you can just keep trying to pull someone off the meatspike and kill them in a few seconds.